### PR TITLE
CSB-6 Log during model training

### PR DESCRIPTION
**Pull request by Noxxxxious**
Sample logs

image

### CODE REVIEW

File: pl-csb-model/__main__.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
 import argparse
-import sacrebleu
```
Comment (Noxxxxious) [inline pl-csb-model/__main__.py:2]: This import was unused

File: pl-csb-model/__main__.py [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+from logging import Logger
```
Comment (filip-szweda) [inline pl-csb-model/__main__.py:2]: Add newline after this import.

File: pl-csb-model/train/model_finetuner.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import gc
+from logging import Logger
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:2]: Group imports.

File: pl-csb-model/train/model_finetuner.py [hunk 4]
```diff
@@ -9,73 +10,91 @@
...
-        item = data.iloc[random.randint(0, len(data)-1)]
-        xx.append(item[l1])
-        yy.append(item[l2])
-    return xx, yy, long1, long2
+    def __init__(self, logger) -> None:
+        self.__logger = logger
 
-def train(model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, data: pd.DataFrame, optimizer: Adafactor, config: ConfigParser) -> None:
-    train_conf = config["TRAINING"]
+    def __cleanup(self) -> None:
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:19]: Can `collect` or `empty_cache` throw any exceptions? If so, catch them.

File: pl-csb-model/train/model_finetuner.py [hunk 5]
```diff
@@ -9,73 +10,91 @@
...
-        xx, yy, lang1, lang2 = get_random_language_pairs(int(train_conf["batch_size"]), LANGS, data)
-        try:
-            tokenizer.src_lang = lang1
-            x = tokenizer(xx, return_tensors='pt', padding=True, truncation=True, max_length=int(train_conf["max_length"])).to(model.device)
-            tokenizer.src_lang = lang2
-            y = tokenizer(yy, return_tensors='pt', padding=True, truncation=True, max_length=int(train_conf["max_length"])).to(model.device)
-            y.input_ids[y.input_ids == tokenizer.pad_token_id] = -100
+        LANGS = [("pl", "pol_Latn"), ("csb", "csb_Latn")]
+
+        self.__logger.info("Starting the training process")
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:48]: Log training configuration.

File: pl-csb-model/train/model_finetuner.py [hunk 6]
```diff
@@ -9,73 +10,91 @@
...
-    optimizer = Adafactor(
-        [p for p in model.parameters() if p.requires_grad],
-        scale_parameter=False,
-        relative_step=False,
-        lr=1e-4,
-        clip_threshold=1.0,
-        weight_decay=1e-3,
-    )
+    def finetune(self, model: AutoModelForSeq2SeqLM, data: pd.DataFrame, tokenizer: NllbTokenizer, config: ConfigParser) -> None:
+        if torch.cuda.is_available():
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:87]: Log if CUDA is available and when it is not.

File: pl-csb-model/train/model_finetuner.py [hunk 7]
```diff
@@ -9,73 +10,91 @@
...
+            model.cuda()
+        
+        optimizer = Adafactor(
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:90]: Can `Adafactor` throw any exceptions? If so, catch them.

**Comment by filip-szweda** [comment]
Add ticket to the root commit message.

File: pl-csb-model/train/model_finetuner.py (hunk 4 above)
Comment (Noxxxxious) [reply pl-csb-model/train/model_finetuner.py:19]: No, neither gc.collect() nor torch.cuda.empty_cache() are likely to throw exceptions under normal circumstances.

File: pl-csb-model/train/model_finetuner.py [hunk 8]
```diff
@@ -38,6 +39,16 @@ def __get_random_language_pairs(self, batch_size, langs, data):
             raise
 
     def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, data: pd.DataFrame, optimizer: Adafactor, config: ConfigParser) -> None:
+        self.__logger.info("=" * 40)
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:42]: Move to a helper method e.g. `__log_train_config`.

File: pl-csb-model/train/model_finetuner.py [hunk 9]
```diff
@@ -1,5 +1,7 @@
 import gc
+from logging import Logger
```
Comment (filip-szweda) [inline pl-csb-model/train/model_finetuner.py:2]: Move after `import random`.

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**